teps/0127-larger-results-via-sidecar-logs.md

```diff
@@ -0,0 +1,293 @@
+
+2. Addressing other [alternatives][tep-0086-alts] for larger `Results` that are listed in [TEP-0086][tep-0086].
+
+### Use Cases
+
+1. Support signing `Results` using SPIRE for non-falsifiable provenance that's required for [SLSA L3][L3]. As described
+   in [TEP-0089][tep-0089], the signatures and certificates used to verify `Results` are stored alongside the `Results`
+   in the `Termination Message`. This exacerbates the size limit issues for `Results`. The size of the certificate is
+   800 bytes and the size of signatures is approximately 100 bytes per `Result`.
+   
+   > For now, signatures of the `Results` will be contained within the `Termination Message` of the `Pod`, alongside
+     any additional material required to perform verification. One consideration of this is the size of the additional
+     fields required. The size of the certificate needed for verification is about 800 bytes, and the size of the
+     signatures is about 100 bytes * (number of `Results` + 1). The current `Termination Message` size is 4K, but there
+     is TEP-0086 looking at supporting larger results.
+
+2. Support emitting structured `Results`. For example, store the images produced by `ko` and all their copies to the
+   various regional registries in a `TaskRun`. The release `Pipeline` has this use case, and it ran into the current
+   size limit of 4096 bytes. As described in the [issue][4282], the images produced were 4572 bytes which didn't fit
+   into `Results`. For further details about structured `Results`, see [TEP-0075][tep-0075] and [TEP-0076][tep-0076].
+
+## Proposal
+
+To support larger `Results`, we propose using stdout logs from a dedicated `Sidecar` to return a json `Result` object.
+The `Pipeline` controller would wait for the `Sidecar` to exit and then read the logs based on a particular query and
+append `Results` to the `TaskRun`.
+
+The dedicated `Sidecar` will be injected alongside other `Steps`. The `Sidecar` will watch the `Results` paths of the
+`Steps`. This `Sidecar` will output the name of the `Result` and its contents to stdout in a parsable pattern. The
+`TaskRun` controller will access the stdout logs of the `Sidecar` then extract the `Results` and its contents.
+
+After the `Steps` have terminated, the `TaskRun` controller will write the `Results` from the logs of the `Sidecar`
+instead of using the `Termination Message`, hence bypassing the 4KB limit. This approach keeps the rest of the existing
+functionality the same and does not require any external storage mechanism.
+
+For further details, see the [demonstration][demo] of the [implementation][poc].
+
+This proposal provides an opportunity to experiment with this solution to provide `Results` within the CRDs as we
+continue to explore other alternatives, including those that involve external storage.
+
+### Feature Gate
+
+This feature will be gated using a `result-extraction-method` feature flag. 
```

Comment:
NIT: may I suggest something simpler, like `result-storage` or `results-from`?

Reply:
`results-from` is much better name...updated, thank you!